### PR TITLE
Feature/issue181

### PR DESCRIPTION
**Pull request by chrisala**

### CODE REVIEW

File: grails-app/services/au/org/ala/ecodata/forms/UserInfoService.groovy [hunk 1]
```diff
@@ -17,27 +26,31 @@ import org.springframework.http.HttpStatus
...
      * @param username username
      * @param key mobile auth key
-     * @return userdetails
+     * @return Map
      *
      **/
-
-    def Map getUserFromAuthKey(String username, String key) {
+    Map getUserFromAuthKey(String username, String key) {
         String url = grailsApplication.config.mobile.auth.check.url
```
Comment (salomon-j) [inline grails-app/services/au/org/ala/ecodata/forms/UserInfoService.groovy:54]: replace to getProperty

File: src/test/groovy/au/org/ala/ecodata/forms/UserInfoServiceSpec.groovy [hunk 2]
```diff
@@ -65,23 +73,86 @@ class UserInfoServiceSpec extends Specification implements ServiceUnitTest<UserI
...
+        when: "Authorization header is  not set and authKeyEnabled is true"
+        grailsApplication.config.mobile.authKeyEnabled = true
+        request.removeHeader('Authorization')
+        request.addHeader(UserInfoService.AUTH_KEY_HEADER_FIELD, key)
+        request.addHeader(UserInfoService.USER_NAME_HEADER_FIELD, userName)
+        result = service.getCurrentUser()
+
+        then:
+        1 * authService.userDetails() >> null
+        1 * webService.doPostWithParams(grailsApplication.config.mobile.auth.check.url, [userName: userName, authKey: key]) >> [ statusCode: 200, resp: [status: "success"]]
```
Comment (salomon-j) [inline src/test/groovy/au/org/ala/ecodata/forms/UserInfoServiceSpec.groovy:150]: replace to getProperty

File: grails-app/services/au/org/ala/ecodata/forms/UserInfoService.groovy (hunk 1 above)
Comment (chrisala) [reply grails-app/services/au/org/ala/ecodata/forms/UserInfoService.groovy]: Thanks Joseph, resolved.

File: src/test/groovy/au/org/ala/ecodata/forms/UserInfoServiceSpec.groovy (hunk 2 above)
Comment (chrisala) [reply src/test/groovy/au/org/ala/ecodata/forms/UserInfoServiceSpec.groovy:150]: Thanks Joseph, resolved.

**Pull request merged**